src/backend/codegen/codegen_wrapper.cc

```diff
@@ -0,0 +1,85 @@
+//---------------------------------------------------------------------------
+//  Greenplum Database
+//  Copyright (C) 2016 Pivotal Software, Inc.
+//
+//  @filename:
+//    init_codegen.cpp
+//
+//  @doc:
+//    C wrappers for initialization of code generator.
+//
+//---------------------------------------------------------------------------
+
+#include "codegen/codegen_wrapper.h"
+#include "codegen/codegen_manager.h"
+#include "codegen/slot_deform_tuple_codegen.h"
+
+#include "codegen/utils/codegen_utils.h"
+
+using gpcodegen::CodeGenManager;
+using gpcodegen::BaseCodeGen;
+using gpcodegen::SlotDeformTupleCodeGen;
+
+// Current code generator manager that oversees all code generators
+static void* ActiveCodeGeneratorManager = nullptr;
+static bool is_codegen_initalized = false;
+
+// Perform global set-up tasks for code generation. Returns 0 on
+// success, nonzero on error.
+int InitCodeGen() {
```

Comment:
I prefer to change all bool to int like @hardikar  suggested. @foyzur  do you agree.